File: Modul 2/first-task.py
import emoji

print('\n<<<Kegiatan 1>>>\n')

redHeart = emoji.emojize(':red_heart:', variant='emoji_type')

words = ['I', redHeart, 'PYTHON', 'AND', 'JAVASCRIPT']

print('<<<Map>>>\n')
def iterateWords(wordList):
    newListWord = []
    for word in wordList:
        newListWord.append(word)
    return newListWord

print('With Pure Function')
print(iterateWords(words))

print('With Map Function')
print(list(map(lambda word: word, words)))

print('\n<<<Filter>>>\n')
def filterWords(wordList):
    newListWord = []
    for word in wordList:
        if 'A' in word:
            newListWord.append(word)
    return newListWord

print('With Pure Function')
print(filterWords(words))

print('With Filter Function')
print(list(filter(lambda word: 'A' in  word, words)))

#Robb - Mobile & Web Enthusiast

File: Modul 2/first-task.js
console.log('\n<<<Kegiatan 1>>>\n');

const words = ['I', '❤️', 'PYTHON', 'AND', 'JAVASCRIPT'];

console.log('<<<MAP>>>\n');
function iterateWords(wordList) {
    const newWordList = [];
    for (const word of wordList) {
        newWordList.push(word);
    }
    return newWordList;
}

console.log('With Pure Function');
console.log(iterateWords(words));

console.log('With Map Function');
console.log(words.map((word) => word));

console.log('\n<<<Filter>>>\n')
function filterWords(wordList) {
    const newWordList = [];
    for (const word of wordList) {
        if (word.includes('A')) {
            newWordList.push(word);
        }
    }
    return newWordList;
}

console.log('With Pure Function');
console.log(filterWords(words));

console.log('With Filter Function');
console.log(words.filter((word) => word.includes('A')));

//Robb - Mobile & Web Enthusiast
